### ISSUE TITLE
Clicking on Hide Button Label in header menu doesn't work as expected

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/SmartTemplate4#304]
Hide button label should hide the label of the header action menu immediately. I think it doesn't work as expected...

**Comment by RealRaven2000** [comment]
The line which determines current status (to be toggled) is only called once (when setting up the menu)

```js
let isLabelHidden = (await messenger.LegacyPrefs.getPref("extensions.smartTemplate4.toolbar.hideLabel"));
```

it needs to be updated every time the menu item is clicked.

**Comment by RealRaven2000** [comment]
Fixed:

[smartTemplate-fx-4.6.2pre4.zip](https://github.com/user-attachments/files/16322518/smartTemplate-fx-4.6.2pre4.zip)

I had to update the status during the event handler, which also required making it async

----

To try the version above download zip file and drag the file into Thunderbird Add-ons Manager (do not extract contents, it won't install like that)

**Comment by RealRaven2000** [comment]
Fixed in 4.7 - published 05/08/2024
